Sanitize the input array length

nemanja-boric-sociomantic opened the pull request:
FiberSelectReader now will throw an exception if the array
length is larger than FiberSelectReader.max_array_size (defaults
to 10MB). This will prevent allocation of the huge amounts of memory
on the invalid data.

gavin-norman-sociomantic reviewed (changes requested):
Shouldn't update beaver and makd, or?

nemanja-boric-sociomantic commented:
Whops, that's a mistake.

nemanja-boric-sociomantic committed:
Sanitize the input array length

FiberSelectReader now will throw an exception if the array
length is larger than FiberSelectReader.max_array_size (defaults
to 10MB). This will prevent allocation of the huge amounts of memory
on the invalid data.

nemanja-boric-sociomantic commented:
Updated.
